File: application/src/main/java/com/smtm/application/users/endpoint/v1/UserProfileDto.java
package com.smtm.application.users.endpoint.v1;

import com.smtm.users.registration.UserProfile;

public class UserProfileDto {

    private final long id;
    private final String email;

    public static UserProfileDto of(UserProfile.Valid createdUser) {
        return of(createdUser.getId(), createdUser.getEmail().toString());
    }

    public static UserProfileDto of(long id, String email) {
        return new UserProfileDto(id, email);
    }

    private UserProfileDto(long id, String email) {
        this.id = id;
        this.email = email;
    }

    public long getId() {
        return id;
    }

    public String getEmail() {
        return email;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        UserProfileDto that = (UserProfileDto) o;

        if (id != that.id) return false;
        return email.equals(that.email);
    }

    @Override
    public int hashCode() {
        int result = (int) (id ^ (id >>> 32));
        result = 31 * result + email.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "UserProfileDto{" +
            "id=" + id +
            ", email='" + email + '\'' +
            '}';
    }
}

File: application/src/main/kotlin/com/smtm/application/security/v1/UserProfileDto.kt
package com.smtm.application.security.v1

import com.smtm.security.registration.UserProfile.Valid

data class UserProfileDto(val id: Long, val email: String) {

    companion object {

        const val MediaTypeValue = "application/smtm.user-profile.v1+json"
    }
}

fun userProfileDtoOf(createdUser: Valid) = userProfileDtoOf(createdUser.id, createdUser.email.toString())

fun userProfileDtoOf(id: Long, email: String) = UserProfileDto(id, email)
